source/tones.py

```diff
@@ -38,18 +39,42 @@ def terminate():
 	global player
 	player = None
 
-def beep(hz,length,left=50,right=50):
+
+#: Notifies when a beep is about to be generated and played,
+#: and allows components or add-ons to decide whether the beep should actually be played.
+#: For example, when controlling a remote system,
+#: the remote system must be notified of beeps played on the local system.
+#: Also, registrars should be able to suppress playing beeps if desired.
+#: Handlers are called with the same arguments as L{beep} as keyword arguments.
+decide_beep = extensionPoints.Decider()
+
+
+def beep(
+		hz: float,
+		length: int,
+		left: int = 50,
+		right: int = 50,
+		isSPeechBeepCommand: bool = False

```

Comment:
there's a typo in this parameter (capital P in speech)
```suggestion
		isSpeechBeepCommand: bool = False
```